src/node/blockstorage.cpp

```diff
@@ -259,8 +259,13 @@ bool BlockManager::LoadBlockIndex()
     std::sort(vSortedByHeight.begin(), vSortedByHeight.end(),
               CBlockIndexHeightOnlyComparator());
 
+    CBlockIndex* previous_index{nullptr};
     for (CBlockIndex* pindex : vSortedByHeight) {
         if (m_interrupt) return false;
+        if (previous_index && pindex->nHeight > previous_index->nHeight + 1) {
+            return error("%s: block index is non-contiguous, index of height %d missing", __func__, previous_index->nHeight + 1);
+        }
+        previous_index = pindex;
```

Comment:
In ad66ca1e:

What if the genesis block index is missing?
The first round of the loop would set `previous_index` to block_index_1. Second one to block_index_2, etc. Making `LoadBlockIndex()` pass.

Reply:
That is true, but the specific case of a missing (or incorrect) genesis block index should get caught in an extra check right after, [see here](https://github.com/bitcoin/bitcoin/blob/c6a338b67e8e7848e6f42329a8b0bf3add16ad51/src/node/chainstate.cpp#L69C1-L74).

There are other cases that wouldn't be caught with the proposed solution: We could have the valid chain and also some extra block indices that don't connect. 
A stricter check would therefore be to require each block index of height `x` (except genesis) to have a predecessor of height `x - 1`. I didn't add it because these kind of extra blocks wouldn't actually hurt anything, plus maybe there are some of these block indices out there because of some bug in some ancient version of core.